### PR TITLE
[expo-splash-screen] add polyfill for usage within managed workflow

### LINKED ISSUE #8067: Base workflow expo-splash-screen error
### Environment
  Expo CLI 3.19.2 environment info:
    System:
      OS: macOS Mojave 10.14.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.14.0 - /usr/local/opt/node@12/bin/node
      Yarn: 1.6.0 - /usr/local/bin/yarn
      npm: 6.13.4 - /usr/local/opt/node@12/bin/npm
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~37.0.3 => 37.0.8 
      react: ~16.9.0 => 16.9.0 
      react-native: ~0.61.5 => 0.61.5 
    npmGlobalPackages:
      expo-cli: 3.19.2

Happens on both IOS and Android. Both in expo client and in xcode on a release version.
### Steps to Reproduce
Use expo-splash-screen in any created project

expo init 
    -bare
expo install expo-splash-screen
cd IOS
pod install
cd .. 
yarn expo-splash-screen "#ec03fc"

edit App.js (add the below code)
    import * as SplashScreen from 'expo-splash-screen';

    SplashScreen.preventAutoHideAsync()
    .catch(console.error);
run `expo start`
run either IOS or Android / or run from xcode/android studio

Get the following error, notice the splash screen is the correct pink/purple color. 

### Expected Behavior
Splash screen never goes away. (in my snack you can see i never hid the splash screen).

### Actual Behavior
undefined is not an object (evaluating '_ExpoSplashScreen.default.preventAutoHideAsync')
- node_modules/expo-splash-screen/build/SplashScreen.js:23:7 in preventAutoHideAsync
- node_modules/regenerator-runtime/runtime.js:45:44 in tryCatch

If you print the keys for NativeModulesProxy in ./node_modules/expo-spash-screen/build/ExpoSplashScreen.js you will notice that `ExpoSplashScreen` does not exist and this is what the error is.

### Reproducible Demo

https://snack.expo.io/3zNZ!I0!E

### PR DESCRIPTION
**Pull request by byCedric**
# Why

This polyfills the `expo-splash-screen` to the older `import { SplashScreen } from 'expo';` in managed Expo projects. It should fix some confusion around the splash screen within the Expo client. (fixes #8067, fixes #7749)

# How

Looked at `expo-updates` and `import { SplashScreen } from 'expo';`. This uses the same signature as the newer `expo-splash-screen`, but uses the existing native APIs.

# Test Plan

Copied the built files into a managed project.

### CODE REVIEW

File: packages/expo-splash-screen/README.md [hunk 1]
```diff
@@ -190,7 +190,7 @@ const styles = StyleSheet.create({
 
 ## 💻 Installation in managed Expo projects
 
-[Managed](https://docs.expo.io/versions/latest/introduction/managed-vs-bare/) Expo projects are not supported by this unimodule at this moment. Instead, you should use the `SplashScreen` module that is within the `expo` package (`import { SplashScreen } from 'expo'`).
+[Managed](https://docs.expo.io/versions/latest/introduction/managed-vs-bare/) Expo projects uses the older `SplashScreen` module within the `expo` package (`import { SplashScreen } from 'expo'`).
```
Comment (ide) [inline packages/expo-splash-screen/README.md:193]:
```suggestion
[Managed](https://docs.expo.io/versions/latest/introduction/managed-vs-bare/) Expo projects use the older `SplashScreen` module within the `expo` package (`import { SplashScreen } from 'expo'`).
```

File: packages/expo/src/launch/SplashScreen.expo.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { NativeModules } from 'react-native';
+
+const { ExponentSplashScreen: SplashScreen = {} } = NativeModules;
+
+export function preventAutoHide() {
```
Comment (ide) [inline packages/expo/src/launch/SplashScreen.expo.ts:5]:
```suggestion
export function preventAutoHide(): void {
```

File: packages/expo/src/launch/SplashScreen.expo.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  if (SplashScreen.preventAutoHide) {
+    SplashScreen.preventAutoHide();
+  }
+}
+
+export function hide() {
```
Comment (ide) [inline packages/expo/src/launch/SplashScreen.expo.ts:11]:
```suggestion
export function hide(): void {
```

**Review by ide — APPROVED** [review]
Make sure to rebuild JS files before committing.

File: packages/expo-splash-screen/README.md (hunk 1 above)
Comment (byCedric) [reply packages/expo-splash-screen/README.md:193]: [expo-splash-screen] fix grammar in readme

**Pull request merged**